kedro-datasets/kedro_datasets/pandas/csv_dataset.py

```diff
@@ -188,3 +188,9 @@ def _invalidate_cache(self) -> None:
         """Invalidate underlying filesystem caches."""
         filepath = get_filepath_str(self._filepath, self._protocol)
         self._fs.invalidate_cache(filepath)
+
+    def _preview(self, nrows) -> Dict:
+        filepath = get_filepath_str(self._filepath, self._protocol)
+        df = pd.read_csv(filepath, nrows=nrows)
+
+        return df.to_dict()
```

Comment:
I think you need the exact logic, the first half is for `file`, and the second half maybe is for the `archived` file type I am not sure - but if it can be `_load` then it should be also `_preview.

Ideally, I think the `preview` method is just using `_load` with an extra `nrows` argument and the transformation to a dict. It's a bit tricky to do it since `_load` doesn't take any argument.

This is something I am thinking about

```python
    def _preview(self, nrows=10):
        dataset_copy = self._copy()  # Create a copy so it doesn't contaminate the original dataset
        dataset_copy._load_args["nrows"] = nrows
        data = dataset_copy.load()
        return data.to_dict()
```

Reply:
thanks @noklam, i tried your suggestion locally and it works fine 😄

Reply:
This is exactly the point I was going to make also, thank you for pointing it out @noklam. We definitely want exactly the same logic as is already in `_load` or there will be cases where there is doesn't work correctly.

I have some other modifications to suggest here, but they are do to with something else so I will put in a separate comment. In terms of working with full generality, I like @noklam's solution here.

Reply:
P.S. @Huongg I also have comments on https://github.com/kedro-org/kedro-viz/pull/1288/ - will try and review today.